src/scripts/main.js

```diff
@@ -1,11 +1,21 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(eventName, (el) => {
```

Comment:
The event listener function parameter `el` is not used in the function body. If you intend to use the event object, consider renaming `el` to `event` for clarity and using it within the function if needed.